owner-onboarding-server/src/main.rs

```diff
@@ -426,7 +428,9 @@ async fn main() -> Result<()> {
                 .or(handler_to2_prove_device)
                 .or(handler_to2_device_service_info_ready)
                 .or(handler_to2_device_service_info)
-                .or(handler_to2_done),
+                .or(handler_to2_done)
+                .or(ov_filter(user_data.clone(),user_data.owner_key.clone()))
```

Comment:
If you're already passing in `user_data`, then why are you also cloning the `owner_key` separately?